internal/policy/container/has_modified_files.go

```diff
@@ -45,118 +48,155 @@ func (p *HasModifiedFilesCheck) Validate(ctx context.Context, imgRef image.Image
 // installed via packages from the container image,and the list of files (packageFilesRef.LayerFiles)
 // modified/added via layers in the image.
 func (p *HasModifiedFilesCheck) getDataToValidate(ctx context.Context, imgRef image.ImageReference) (*packageFilesRef, error) {
-	logger := logr.FromContextOrDiscard(ctx)
-
-	// Get a list of packages from the RPM database. This avoids having to rely on
-	// rpm, dnf, yum, etc. being installed in the image.
-	pkgList, err := rpm.GetPackageList(ctx, imgRef.ImageFSPath)
+	fs := afero.NewOsFs()
+	layerDir, err := afero.TempDir(fs, "", "rpm-layers-*")
+	defer func() {
```

Comment:
Two Items here:

---
Bit of a nit

Let's move this deferral below the error checking.

Logically, I know that fs.RemoveAll does not return an error if the underlying fs doesn't exist. But layerDir and err are assigned together, and I will always do a double-take if the assigned variable is used before the error assigned in the same line is checked.

---
I'm sure golangci-lint told you to explicitly ignore this error using `_`, but I would probably justify ignoring that lint enforcement here. It's not too uncommon to do what you had before `defer thingThatMightErrorButWeDoNotCare()`

Reply:
> Two Items here:
> 
> Bit of a nit
> 
> Let's move this deferral below the error checking.
> 
> Logically, I know that fs.RemoveAll does not return an error if the underlying fs doesn't exist. But layerDir and err are assigned together, and I will always do a double-take if the assigned variable is used before the error assigned in the same line is checked.

linting told me to do that anyway, so it should already be moved.

> 
> I'm sure golangci-lint told you to explicitly ignore this error using `_`, but I would probably justify ignoring that lint enforcement here. It's not too uncommon to do what you had before `defer thingThatMightErrorButWeDoNotCare()`

I actually do prefer the explicit ignoring of the error. It makes it clear that the error is being ignored on purpose, rather than just being an oversight.